### ISSUE TITLE
support for non-http/https schemes

### ISSUE THREAD

**Issue by wzshiming** [swagger-api/swagger-ui#5220]
any protocol (such as ws/wss) scheme can be specified in the `servers/url` property.

**Comment by shockey** [comment]
@wzshiming, what protocol(s) are you seeking support for?

Generally, we're pretty limited in what we can offer in Swagger UI because of restrictions in browser support, as well as the Fetch/XmlHttpRequest standards.

**Comment by wzshiming** [comment]
Such as the WebSocket (ws/wss)

**Comment by shockey** [comment]
I've catalogued this as a feature request!

I'm not sure when/if we'll support this, since (1) Swagger UI is largely designed around a 1:1 request/response flow, and (2) `window.fetch`, which we use for network requests, [doesn't support ws/wss](https://github.com/github/fetch/issues/499).

**Comment by nithinbandaru1** [comment]
@shockey

According to this URL [https://swagger.io/docs/specification/api-host-and-base-path/](https://swagger.io/docs/specification/api-host-and-base-path/) it is mentioned that ws and wss can be specified in URL scheme.

So, I assume web sockets can be represented using swagger UI.
I think we can use same paths region to specify what will be request and what will be response. Only thing I need to specify is direction if request is from server or from client.

Is there any example for this??

**Comment by pyrofolium** [comment]
yeah mentioned and no examples last I looked. I'm going to assume it's not supported.
